google/auth/external_account.py

```diff
@@ -161,13 +166,25 @@ def info(self):
                 useful for serializing the current credentials so it can deserialized
                 later.
         """
-        config_info = {
-            "type": _EXTERNAL_ACCOUNT_JSON_TYPE,
+        config_info = self._constructor_args()
+        config_info.update(
+            type=_EXTERNAL_ACCOUNT_JSON_TYPE,
+            service_account_impersonation=config_info.pop(
+                "service_account_impersonation_options", None
+            ),
+        )
+        config_info.pop("scopes", None)
+        config_info.pop("default_scopes", None)
+        return {key: value for key, value in config_info.items() if value is not None}
+
+    def _constructor_args(self):
+        args = {
             "audience": self._audience,
             "subject_token_type": self._subject_token_type,
             "token_url": self._token_url,
+            "token_info_url": self._token_info_url,
             "service_account_impersonation_url": self._service_account_impersonation_url,
-            "service_account_impersonation": copy.deepcopy(
+            "service_account_impersonation_options": copy.deepcopy(
```

Comment:
why is this called service_account_impersonation sometimes and service_account_impersonation_options sometimes?

Reply:
I'm not sure.  I don't mind leaving this as service_account_impersonation_options, but I also didn't want to break anybody's code using our API

Reply:
Looks like this is existing behavior. So I am not blocking for this PR. But would be good to follow up with if we can standardize this. I will let AION SDK team decide

Reply:
I think it has to do with the fact that this is called "service_account_impersonation" in the JSON file.  But yeah, consistent behavior here would be nice